README.md
# nhkradio2-langs-record
NHKラジオ第２ 語学番組録音（汎用版）

- これはNHKラジオ第２放送の「語学ラジオ番組」を**自動録音**するものです。
- 全ての番組の録音に対応しています。

[NHKゴガク：世界にログイン](https://www.nhk.or.jp/gogaku/#indexMainList)  

- ロシア語限定版には [nhkradio2-russian-record](https://github.com/ru-museum/nhkradio2-russian-record) があります。

### NHKラジオ第2 語学番組表(2023-6)

<table>
<tr>
  <th>ID</th><th>番組名</th><th>編</th><th>曜日</th><th>放送時間</th><th>再放送</th>
</tr>  
<tr>
  <td>0</td><td>まいにちロシア語</td><td nowrap>初級</td><td>月〜水</td><td>8:50 - 9:05</td><td>月〜水 16:30 - 45／翌週 月〜水 15:30 - 45</td>
</tr>
<tr>
  <td></td><td></td><td>応用</td><td>木・金 </td><td>8:50 - 9:05</td><td>木・金 16:30 - 45／翌週 木・金 15:30 - 45</td>
</tr>
<tr>
  <td>1</td><td>まいにちフランス語</td><td>初級</td><td>月〜水</td><td>7:30 - 45</td><td>月〜水 14:30 - 45／翌週 月〜水 11:00 - 15</td>
</tr>
<tr>
  <td></td><td></td><td>応用</td><td>木・金</td><td>7:30 - 45</td><td>木・金 14:30 - 45／翌週 木・金 11:00 - 15</td>
</tr>
<tr>
  <td>2</td><td>まいにちドイツ語</td><td>初級</td><td>月〜水</td><td>7:00 - 15</td><td>月〜水 15:15 - 30／翌週 月〜水 11:30 - 45
</td>
</tr>
<tr>
  <td></td><td></td><td>応用</td><td>木・金</td><td>7:00 - 15</td><td>木・金 15:15 - 30／翌週 木・金 11:30 - 45</td>
</tr>  
<tr>
  <td>3</td><td>まいにちイタリア語</td><td>初級</td><td>月〜水</td><td>7:45 - 8:00</td><td>月〜水 16:45 - 17:00／翌週 月〜水 11:15 - 30</td>
</tr>
<tr>
  <td></td><td></td><td>応用</td><td>木・金</td><td>7:45 - 8:00</td><td>木・金 16:45 - 17:00／翌週 木・金 11:15 - 30</td>
</tr> 
<tr>
  <td>4</td><td>まいにちスペイン語</td><td>初級</td><td>月〜水</td><td>7:15 - 30</td><td>月〜水 14:45 - 15:00／翌週 月〜水 11:45 - 12:00</td>
</tr>
<tr>
  <td></td><td></td><td>応用</td><td>木・金</td><td>7:15 - 30</td><td>木・金 14:45 - 15:00／翌週 木・金 11:45 - 12:00</td>
</tr>
<tr>
  <td>5</td><td>まいにち中国語</td><td></td><td>月〜金</td><td>8:15 - 30</td><td>月〜金 10:15 - 30／日 11:00 - 12:15（5回分）</td>
</tr>
<tr>
  <td>6</td><td>ステップアップ中国語</td><td></td><td>月・火</td><td>10:30 - 45</td><td>月・火 22:00 - 15／日 9:00 - 30（2回分）</td>
</tr>
<tr>
  <td>7</td><td>小学生の基礎英語</td><td></td><td>月・水・金</td><td>18:35 - 45</td><td>火・木 18:35 - 45（前日の再放送）<br>／土 6:00 - 30（3回分）、14:10 - 40（3回分）<br>／日曜 8:00 - 30（3回分）、21:00 - 30（3回分）</td>
</tr>
<tr>
  <td>8</td><td>中学生の基礎英語 レベル1</td><td></td><td>月〜金</td><td>6:00 - 15</td><td>月〜金 18:45 - 19:00、21:00 - 15<br>／土 21:00 - 22:15（5回分）
</td>
</tr>
<tr>
  <td>9</td><td>中学生の基礎英語 レベル2</td><td></td><td>月〜金</td><td>6:15 - 30</td><td>月〜金 19:00 - 15、21:15 - 30<br>／土 22:15 - 23:30（5回分）</td>
</tr>
<tr>
  <td>10</td><td>中高生の基礎英語 in English</td><td></td><td>月〜金</td><td>6:30 - 45</td><td>月〜金 19：15 - 30、21：30 - 45<br>（第1週・第2週は新作、第3週・第5週は第1週の<br>再放送、第4週は第2週の再放送）</td>
</tr>
<tr>
  <td>11</td><td>ラジオ英会話</td><td></td><td>月〜金</td><td>6:45 - 7:00</td><td>月〜金 12:25 - 40、21:45 - 22:00<br>／日 16:30 - 17:45（5回分）
</td>
</tr>
<tr>
  <td>12</td><td>ボキャブライダー</td><td></td><td>月〜金</td><td>9:05 - 10</td><td>月〜金 13:30 - 35、16:20 -25、23:35 - 40<br>／土 6:30 - 6:55（5回分）、12:15 - 40（5回分）</td>
</tr>
<tr>
  <td>13</td><td>エンジョイ・シンプル<br>・イングリッシュ</td><td></td><td>月〜金</td><td>9:10 - 15</td><td>月〜金 12:55 - 13:00、16:25 - 30、23:10 - 15<br>／土 9:05 - 30（5回分）<br>／日 14:20 - 45（5回分）、21:30 - 55（5回分）</td>
</tr>
<tr>
  <td>14</td><td>英会話タイムトライアル</td><td></td><td>月〜金</td><td>8:30 - 40</td><td>月〜金 12:15 - 25、18:25 - 35、23:00 - 10<br>／土 7:00 - 50（5回分）</td>
</tr>
<tr>
  <td>15</td><td>ラジオビジネス英語</td><td></td><td>月〜金</td><td>7:45 - 8:00</td><td>月〜金 9:15 - 30、12:40 - 55、23:20 - 35<br>／土 10:30 - 11:45（5回分）
</td>
</tr>
<tr>
  <td>16</td><td>ニュースで学ぶ「現代英語」</td><td></td><td>月〜金</td><td>9:30 - 45</td><td>月〜金 15:45 - 16:00／日 22:00 - 23:15（5回分）</td>
</tr>
<tr>
  <td>17</td><td>まいにちハングル講座</td><td></td><td>月〜金</td><td>8:00 - 15</td><td>月〜金 10:45 - 11:00、18:10 - 25<br>／日 14:45 - 16:00（5回分）
</td>
</tr>
<tr>
  <td>18</td><td>ステップアップ ハングル講座</td><td></td><td>木・金</td><td>10:30-10:45</td><td>木・金 22:00 - 15／土・日 17:45 - 18:00
</td>
</tr>
<tr>
  <td>19</td><td>アラビア語講座</td><td></td><td>日</td><td>9:30 - 10:00</td><td></td>
</tr>
 <tr>
  <td>20</td><td>ポルトガル語入門</td><td></td><td>土</td><td>18:45 - 19:00</td><td>日 午後11:30-11:45<br>／翌週 土 8:50 - 9:05、14:40 - 14:55
</td>
</tr>
</table>

# 動作環境  
- Linux（Debian: **bash** 使用）
- ffmpeg のインストールが必要です。

# 録音方法  
1. 作業フォルダ（設定例「NHKラジオ語学講座」）に nhkradio2-langs-record.sh を置き編集します。  

- 必要あれば講座に応じた録音時間を指定します（参照：録音開始時間の微調整）。
- **講座を複数録音するに際し修正を加えた場合には、各講座毎に別ファイルで実行させて下さい。**  
　【例】 nhkradio2-**french**-record.sh -i 1
 ```
# 録音時間設定値 ： 通常15分
REC_TIME="00:15:00" 
 ```
2. nhkradio2-langs-record.sh にアクセス権の実行を許可します。  
 ```
 # chmod 755 nhkradio2-langs-record.sh
 ```
3. 作業フォルダでヘルプを表示させ録音したい**講座 ID 番号**を確認します。  
　【例】**0** まいにちロシア語　...  
　　　　**1** まいにちフランス語　...  
 ※ sh ではなく **bash** を使用して下さい。
 ```
 # bash ./nhkradio2-langs-record.sh -h
 ``` 
4. CRON に講座に応じた放送曜日と時刻を登録します。  
- 設定ファイル： /etc/crontab  
- **CRON への登録は講座毎に必要です。**  

**【設定例】**  
　登録講座： 0 まいにちロシア語：月〜水【初級編】/ 木・金【応用編】8:50-9:05  
- ファイルのオプション引数として講座ID番号を指定します。  
- **入門編及び応用編のある講座で何れかのみを録音したい場合は「曜日指定」で設定します。**  
 

<table>
<tr>
  <td>オプション指定</td><td>-i 0（講座番号）</td>
</tr>  
<tr>
  <td>録音時刻指定</td><td>50 8（8時50分）</td>
</tr>
<tr>
  <td>曜日指定</td><td>1-5（月〜金）<br>初級編のみ：1-3（月〜水）<br>応用編のみ：4,5（木・金）</td>
</tr>
</table>

<table>
<tr>
  <th colspan="7">曜日対応表</th>
</tr>
<tr>
  <td>日</td><td>月</td><td>火</td><td>水</td><td>木</td><td>金</td><td>土</td>
</tr>
<tr>
  <td>0</td><td>1</td><td>2</td><td>3</td><td>4</td><td>5</td><td>6</td>
</tr>
</table>

- **username** は、root 或いは使用するユーザー名です（無いとエラーが出ます）。  
- 作業フォルダを「NHKラジオ語学講座」に設定した場合です。  
　**/your/directory/name/NHKラジオ語学講座/** は**絶対 PATH**で指定します。  
  ※ シェルは **bash** を指定して下さい。
 ```
50 8 * * 1-5 username bash /your/directory/name/NHKラジオ語学講座/nhkradio2-langs-record.sh -i 0
 ```
5. CRON を再起動させます。
 ```
 # /etc/init.d/cron restart
 ```
 <br> 
※ スクリプトが起動されると「作業フォルダ」に参照用ファイル（languages.csv：番組リスト）が作成されます。
  <br>
  <br>
  
## 録音開始時間の微調整  
- 回線状況や配信（放送時刻）とPCの時計の時刻が正確に合致していない場合など、録音開始時間に**誤差**の生じる場合があります。  
- 当方の PC の内蔵時計では日本標準時との誤差は 0.2 秒でしたが、**36秒**の遅延調整が必要でした。  
　　[情報通信研究機構/日本標準時](https://www.nict.go.jp/JST/JST5.html)
- **オプションで開始時刻遅延を行った場合には、別ファイルにする必要はありません。**  

**(A)** 開始時刻を遅らせる場合：録音開始時間調整の　**sleep** の値を変更します（初期値：0）。  

　【例】開始時刻を**36秒**遅らせる  
- オプションで指定する場合：-s **36**  
　　……/nhkradio2-langs-record.sh -i 0 **-s 36**  

- 直接書き換える場合：
```
SLPSECONDS=36 # 開始時刻遅延初期値： 秒

或いは、

sleep 36　  
```
**(B)** 開始時刻を早める場合（稀なケース）： CRON で行います（CRON の再起動が必要）。  
　例：録音開始時刻を**20秒**早める：開始時刻を**1分**早め（初期値：50 ⇒ 49）開始時刻まで**40秒** sleep させます。
```
sleep 40

# CRON
49 8 * * 1-5 sleep 40; username bash /your/directory/name/NHKラジオ語学講座/nhkradio2-langs-record.sh -i 0  
```

## 実動作前のテスト  
1. 録音時間を**10秒**程度に設定します。
 ```
# 録音時間設定値 ： 10秒
REC_TIME="00:00:10" 
 ```
2.  CRON の起動時刻を数分後に設定し再起動します。
 ```
# CRON
49 8 * * 1-5 ……（略）
 ```
3. 「作業フォルダ」に講座名のフォルダと講座名の録音ファイルが保存されているかを確認して下さい。

## 録音ファイル  

- 保存された **m4a** は音声データフォーマットであり、iTunes や通常のプレイヤーで視聴出来ます。 

# 視聴アプリ  
- 初級編及び応用編のある**ロシア語**、**フランス語**、**ドイツ語**、**イタリア語**、**スペイン語**を視聴出来るアプリを公開しました。
- データは編集可能となっています。   
- **ロシア語限定版**はこちらにあります。[nhkradio2-russian-record](https://github.com/ru-museum/nhkradio2-russian-record)

## 　Demo
- 以下で公開用テンプレートをご覧頂けます（**/docs** を参照しています) 。
- サンプルのダミーファイル（【初級編】第１課、【応用編】第1課）を試聴出来ます。 
- 操作方法等はアプリの「視聴方法」に詳しく解説しています。
- 2023年度10月からの新シリーズに対応しています。
- **セキュリティ上ダウンロード後でなくてはデータは読み込み出来ません**。   
  
　　[GitHub Pages: nhkradio2-langs-record](https://ru-museum.github.io/nhkradio2-langs-record/)  
　　[https://ru-museum.github.io/nhkradio2-langs-record/](https://ru-museum.github.io/nhkradio2-langs-record/)  

# 注意  
- 配信 URL は時期は不明ですが変更されています。  
旧URL: "https://nhkradioakr2-i.akamaihd.net/hls/live/511929/1-r2/1-r2-01.m3u8"  
新URL: "https://radio-stream.nhk.jp/hls/live/2023501/nhkradiruakr2/master.m3u8"
- **録音データは著作権上私的利用のみに限定されていますのでご注意下さい。**




CHANGELOG.md


# nhkradio2-langs-record CHANGELOG

2023-11-03
- 視聴アプリを更新しました。  

2023-06-13
- 番組の変更に伴い削除・修正を行いました。  

2021-09-22
- 録音のズレ修正の為の「録音開始時間の微調整」をオプション指定で可能にしました（README 参照）。  
　【例】開始時刻を**36秒**遅らせる  
　　オプションで指定する場合：-s **36**  
　　……/nhkradio2-langs-record.sh -i 0 **-s 36**  
 **※ オプションで開始時刻遅延を行った場合には、別ファイルにする必要はありません。**


nhkradio2-langs-record.sh
#!/bin/bash

set -euo pipefail

# --------------------------------------------------
# 
# NHKラジオ第2放送　語学講座番組録音 - 汎用版　
# 
# --------------------------------------------------
# 放送波URL: 以下で取得出来ます（2021）。 
# https://www.nhk.or.jp/radio/config/config_web.xml

FILECSV="languages.csv"

# 【対応言語講座】
# NHKラジオ語学講座リスト
LANGUAGE_COURSES=(
  "0,まいにちロシア語,初級編 応用編,月〜水 木・金,8:50-9:05,00:15:00,1-2-3 4-5"
  "1,まいにちフランス語,初級編 応用編,月〜水 木・金,7:30-7:45,00:15:00,1-2-3 4-5"
  "2,まいにちドイツ語,初級編 応用編,月〜水 木・金,7:00-7:15,00:15:00,1-2-3 4-5"
  "3,まいにちイタリア語,初級編 応用編,月〜水 木・金,7:45-8:00,00:15:00,1-2-3 4-5"
  "4,まいにちスペイン語,初級編 応用編,月〜水 木・金,7:15-7:30,00:15:00,1-2-3 4-5"
  "5,まいにち中国語,,月〜金,8:15-8:30,00:15:00,1-2-3-4-5"
  "6,ステップアップ中国語,,月・火,10:30-10:45,00:15:00,1-2"
  "7,小学生の基礎英語,,月・水・金,18:35-18:45,00:10:00,1-3-5"
  "8,中学生の基礎英語レベル1,,月〜金,6:00-6:15,00:15:00,1-2-3-4-5"
  "9,中学生の基礎英語レベル2,,月〜金,6:15-6:30,00:15:00,1-2-3-4-5"
  "10,中高生の基礎英語inEnglish,,月〜金,6:30-6:45,00:15:00,1-2-3-4-5"
  "11,ラジオ英会話,,月〜金,6:45-7:00,00:15:00,1-2-3-4-5"
  "12,ボキャブライダー,,月〜金,9:05-9:10,00:05:00,1-2-3-4-5"
  "13,エンジョイ・シンプル・イングリッシュ,,月〜金,9:10-9:15,00:05:00,1-2-3-4-5"
  "14,英会話タイムトライアル,,月〜金,8:30-8:40,00:10:00,1-2-3-4-5"
  "15,ラジオビジネス英語,,月〜金,7:45-8:00,00:15:00,1-2-3-4-5"
  "16,ニュースで学ぶ「現代英語」,,月〜金,9:30-9:45,00:05:00,1-2-3-4-5"
  "17,まいにちハングル講座,,月〜金,8:00-8:15,00:15:00,1-2-3-4-5"
  "18,ステップアップ ハングル講座,,木・金,10:30-10:45,00:15:00,4-5"
  "19,アラビア語講座,,日,9:30-10:00,00:30:00,0"
  "20,ポルトガル語入門,,土,18:45-19:00,00:15:00,6"  
)

# HELP 表示用リスト：ファイルへ出力
cat <<EOM > ${FILECSV}
 0,まいにちロシア語,初級編 応用編,月〜水 木・金,8:50-9:05,00:15:00,1-2-3 4-5
 1,まいにちフランス語,初級編 応用編,月〜水 木・金,7:30-7:45,00:15:00,1-2-3 4-5
 2,まいにちドイツ語,初級編 応用編,月〜水 木・金,7:00-7:15,00:15:00,1-2-3 4-5
 3,まいにちイタリア語,初級編 応用編,月〜水 木・金,7:45-8:00,00:15:00,1-2-3 4-5
 4,まいにちスペイン語,初級編 応用編,月〜水 木・金,7:15-7:30,00:15:00,1-2-3 4-5
 5,まいにち中国語,,月〜金,8:15-8:30,00:15:00,1-2-3-4-5
 6,ステップアップ中国語,,月・火,10:30-10:45,00:15:00,1-2
 7,小学生の基礎英語,,月・水・金,18:35-18:45,00:10:00,1-3-5
 8,中学生の基礎英語レベル1,,月〜金,6:00-6:15,00:15:00,1-2-3-4-5
 9,中学生の基礎英語レベル2,,月〜金,6:15-6:30,00:15:00,1-2-3-4-5
 10,中高生の基礎英語inEnglish,,月〜金,6:30-6:45,00:15:00,1-2-3-4-5
 11,ラジオ英会話,,月〜金,6:45-7:00,00:15:00,1-2-3-4-5
 12,ボキャブライダー,,月〜金,9:05-9:10,00:05:00,1-2-3-4-5
 13,エンジョイ・シンプル・イングリッシュ,,月〜金,9:10-9:15,00:05:00,1-2-3-4-5
 14,英会話タイムトライアル,,月〜金,8:30-8:40,00:10:00,1-2-3-4-5
 15,ラジオビジネス英語,,月〜金,7:45-8:00,00:15:00,1-2-3-4-5
 16,ニュースで学ぶ「現代英語」,,月〜金,9:30-9:45,00:05:00,1-2-3-4-5
 17,まいにちハングル講座,,月〜金,8:00-8:15,00:15:00,1-2-3-4-5
 18,ステップアップ ハングル講座,,木・金,10:30-10:45,00:15:00,4-5
 19,アラビア語講座,,日,9:30-10:00,00:30:00,0
 20,ポルトガル語入門,,土,18:45-19:00,00:15:00,6  
EOM

# このスクリプトファイルが存在する作業ディレクトリ PATH を取得。
SCRIPT_DIR=$(cd $(dirname $0) && pwd)

# 日時生成
arr=("日" "月" "火" "水" "木" "金" "土")
DAY=`date '+%w'`
_DAY="(${arr[$DAY]})"
YEAR=`date '+%Y'`
DATE1=`date '+%m%d'`
DATE2=`date '+%H:%M'`
DATE="${YEAR}${DATE1}${_DAY}-${DATE2}"

# --------------------------------------------------
# 基本変数初期化
# ストリーミング配信 URL
M3U8URL="https://radio-stream.nhk.jp/hls/live/2023501/nhkradiruakr2/master.m3u8"
# 2021年9月現在 URL が変更されています。
# 旧URL：上記で動作しない場合は試して下さい。
# M3U8URL="https://nhkradioakr2-i.akamaihd.net/hls/live/511929/1-r2/1-r2-01.m3u8"
COURSE=""
COURSENUM=""
COURSENAME=""
SLPSECONDS=0 # 開始時刻遅延初期値： 秒 -- 変更可能。

# 番組リストファイル
DATAFILE="${SCRIPT_DIR}/${FILECSV}" # 番組リストファイル

# --------------------------------------------------
# 番組リスト表示：column 6,7 は非表示
function dsp_list {
   echo "-------------------------------------------------------------------------------"
   echo "ID|講座名                              |コース分類   |放送曜日     |放送時間"
   echo "-------------------------------------------------------------------------------"
   cat $DATAFILE | column -H 6,7 -t -s ',' -o '|'
   echo "-------------------------------------------------------------------------------"
   echo "  "
  exit 3
}

# --------------------------------------------------
# helpを作成
function dspHelp {
  cat <<EOM
Usage: $(basename "$0") -i [ID] -c [ p | a ]
  -h            ヘルプを表示します。
  -i [VALUE]    講座 ID 番号を指定します。
  -c [VALUE]    講座が複数(初級編、応用編)あり何れか一方のみを録音する場合：
                初級編 [p] | 応用編 [a]　で指定します。
                ※ オプションを無記入の場合、初級編及び応用編共に録音されます。
  -s [VALUE]    開始時刻を遅延させる： 秒              
                
【NHKラジオ第二放送 語学講座番組】(2023年度6月) *講座のID番号を選択します。
EOM
   dsp_list # 番組リスト表示
   exit 2
}

# --------------------------------------------------
# OPTION 引数別の処理定義
while getopts ":c:i:s:h" optKey; do
  case "$optKey" in
    i)
      COURSENUM="${OPTARG}"  # 指定講座 ID の OPTION 入力値
      ;;
    c)
      COURSENAME="${OPTARG}" # 初級編・応用編の OPTION 入力値
      ;;
    s)
      SLPSECONDS="${OPTARG}" # 開始時刻遅延の OPTION 入力値
      ;;
    '-h'|'--help'|* )
      dspHelp
      ;;
  esac
done

# --------------------------------------------------
# 初級編又は応用編何れかのみに限定する（曜日設定）には CRON( [] 部分 ) で行います。
# 初級編：1-3　応用編：4-5　　【例】 50 8 * * [1-5]	
# ID 0-4 までの講座に付加
if [ ${COURSENUM} -lt 5 ]; then
  if [ $DAY -ge 1 -a $DAY -le 3 ];then
        COURSE="【初級編】"
  elif [ $DAY -eq 4 -o $DAY -eq 5 ];then
        COURSE="【応用編】"
  fi
fi

# --------------------------------------------------
# 講座名の取得
getCourseName () {
    list="${LANGUAGE_COURSES[$1]}" # 指定ID 
    str=(${list//,/ })             # コンマ区切りで配列にする
    echo "${str[1]}"
}

# --------------------------------------------------
# 前処理
# --------------------------------------------------
# 通常は講座名が入ります：変更可
# 指定が無ければ選択された講座名が設定されます。
TITLE=""
if [ "$TITLE" = "" ]; then
   # TITLE="${LANG_COURSES[${COURSENUM},name]}"
   TITLE=`getCourseName $COURSENUM`
fi

# 作業ディレクトリに「講座名」の保存ディレクトリを作成します。
DIRNAME=`getCourseName $COURSENUM`

# ファイルの保存ディレクトリ設定：絶対 PATH を指定 -- 変更可能。
# 通常は /root/ 或いは /home/username/ から始まります
SAVEDIR="${SCRIPT_DIR}/${DIRNAME}"

# 保存ファイル名 ＋ コース ＋ 日付
SAVEFILE="${TITLE}${COURSE}-${DATE}"

dir="${SAVEDIR}" # 一時変数

# 保存ディレクトリ ＋ 保存ファイル名
SAVEFILE_PATH="${dir}/${SAVEFILE}"

# 保存ディレクトリが無ければ作成。
makeDirctory(){
  if [[ ! -e $1 ]]; then
      mkdir $1
  elif [[ ! -d $1 ]]; then
      echo "$1 already exists but is not a directory" 1>&2
  fi
}

# 録音時間設定：　時：分：秒 -- 変更可能。　
# 秒で設定する場合： "900"（15分 x 60秒） 
REC_TIME="00:15:00" 

# 開始時刻を遅延させる： 秒 -- 変更可能。
# sleep 5m 50s （分：秒）
sleep $SLPSECONDS

# 録音したファイルを保存
if [ COURSENUM ]; then

  # 保存ディレクトリが無ければ作成。
  makeDirctory $dir
  
  # 録音して保存
  ffmpeg -i "${M3U8URL}" -t "${REC_TIME}" -c copy "${SAVEFILE_PATH}".m4a

else
  echo "OPTION: -i [ID] で番号を指定して下さい" 1>&2
  exit 1
fi

exit
